src/contents/ui/Full.qml

```diff
@@ -6,16 +6,75 @@ import org.kde.plasma.plasmoid
 import org.kde.plasma.components as PlasmaComponents3
 import org.kde.kirigami as Kirigami
 import org.kde.plasma.private.mpris as Mpris
+import Qt5Compat.GraphicalEffects
 
 
 Item {
     property string albumPlaceholder: plasmoid.configuration.albumPlaceholder
     property real volumeStep: plasmoid.configuration.volumeStep
+    property bool useImageColors: plasmoid.configuration.fullAlbumCoverAsBackground
+    // property int fullViewImplicitMargins: 5
 
     Layout.preferredHeight: column.implicitHeight
     Layout.preferredWidth: column.implicitWidth
     Layout.minimumWidth: column.implicitWidth
     Layout.minimumHeight: column.implicitHeight
+    // Layout.maximumWidth: column.implicitWidth
+    // Layout.maximumHeight: column.implicitHeight
+
+    Kirigami.Theme.backgroundColor: useImageColors ? imageColors.bgColor : Kirigami.Theme.backgroundColor
+    Kirigami.Theme.textColor: useImageColors ? imageColors.fgColor : Kirigami.Theme.textColor
+    Kirigami.Theme.highlightColor: useImageColors ? imageColors.fgHighlightColor : Kirigami.Theme.highlightColor
+
+    Image {
+        visible: useImageColors
+        id: albumArtFull
+        Layout.margins: 0
+        Layout.alignment: Qt.AlignHCenter
+        anchors.horizontalCenter: column.horizontalCenter
+        anchors.verticalCenter: Layout.verticalCenter
+        height: column.height // + fullViewImplicitMargins * 2
+        width: column.width // + fullViewImplicitMargins * 2
+        // x: -fullViewImplicitMargins
+        // y: -fullViewImplicitMargins
+        // z: -1
+        fillMode: Image.PreserveAspectCrop
+        
+        onStatusChanged: {
+            if (status === Image.Ready) {
+                imageColors.update()
+            }
+        }
+
+        source: {
+            if (status === Image.Error || !player.artUrl) {
+                return albumPlaceholder;
+            }
+            return player.artUrl;
+        }
+
+        LinearGradient {
+            id: mask
+            anchors.fill: albumArtFull
+            gradient: Gradient {
+                GradientStop { position: 0; color: "transparent" }
+                GradientStop { position: 0.4; color: "transparent" }
+                GradientStop { position: 0.7; color: imageColors.bgColor }
+                GradientStop { position: 1; color: imageColors.bgColor }
+            }
+        }
```

Comment:
This need to be moved outside the `Image` component, otherwise the `imageColors` of the next album cover will be influenced by the background color of the current album cover.